barman/backup_executor.py

```diff
@@ -1388,6 +1381,197 @@ def _format_src(self, path):
+            )
+
+    def validate_configuration(self):
+        """Verify cofiguration is valid for a snapshot backup."""
+        excluded_config = (
+            "backup_compression",
+            "bandwidth_limit",
+            "network_compression",
+            "tablespace_bandwidth_limit",
+        )
+        for config_var in excluded_config:
+            if getattr(self.server.config, config_var):
+                self.server.config.update_msg_list_and_disable_server(
+                    "%s option is not supported by snapshot backup_method" % config_var
+                )
+
+        if self.config.reuse_backup in ("copy", "link"):
+            self.server.config.update_msg_list_and_disable_server(
+                "reuse_backup option is not supported by snapshot backup_method"
+            )
+
+        required_config = (
+            "snapshot_disks",
+            "snapshot_instance",
+            "snapshot_provider",
+            "snapshot_zone",
+        )
+        for config_var in required_config:
+            if not getattr(self.server.config, config_var):
+                self.server.config.update_msg_list_and_disable_server(
+                    "%s option is required by snapshot backup_method" % config_var
+                )
+
+    @staticmethod
+    def add_mount_data_to_backup_info(backup_info, remote_cmd):
+        for snapshot in backup_info.snapshots_info["snapshots"].values():
+            mount_point, mount_options = remote_cmd.findmnt(snapshot["device"])
+            if mount_point is None:
+                raise BackupException(
+                    "Could not find mount point for device %s" % snapshot["device"]
+                )
+            else:
+                snapshot["mount_point"] = mount_point
+                snapshot["mount_options"] = mount_options
+
+    def backup_copy(self, backup_info):
+        """Actually take a snapshot here"""
+        # Create data dir so backup_label can be written
+        cmd = UnixLocalCommand(path=self.server.path)
+        cmd.create_dir_if_not_exists(backup_info.get_data_directory())
+
+        # Start the snapshot
+        self.copy_start_time = datetime.datetime.now()
+
+        self.snapshot_interface.take_snapshot_backup(
+            backup_info, self.snapshot_instance, self.snapshot_zone, self.snapshot_disks
+        )
+
+        self.copy_end_time = datetime.datetime.now()
+
+        # Gather additional metadata to assist recovery of snapshots
+        remote_cmd = UnixRemoteCommand(ssh_command=self.server.config.ssh_command)
+        self.add_mount_data_to_backup_info(backup_info, remote_cmd)
+
+        # Store statistics about the copy
+        copy_time = total_seconds(self.copy_end_time - self.copy_start_time)
+        backup_info.copy_stats = {
+            "copy_time": copy_time,
+            "total_time": copy_time,
+        }
+
+    @staticmethod
+    def check_instance_exists(snapshot_interface, snapshot_instance, snapshot_zone):
```

Comment:
This doesn't make much sense - if the caller has the snapshot interface, the instance and the zone, why does it not just call snapshot_interface.instance_exists directly?